src/components/ContactDetails/ContactDetailsAvatar.vue

```diff
@@ -330,6 +356,47 @@ export default {
 			}
 		},
 
+		/**
+		 * Downloads the Avatar from social media
+		 *
+		 * @param {String} network the social network to use (or 'any' for first match)
+		 */
+		async getSocialAvatar(network) {
+
+			if (!this.loading) {
+
+				this.loading = true
+				try {
+					const response = await axios.get(generateUrl('/apps/contacts/api/v1/social/avatar/{network}/{id}/{uid}', {
+						network: network.toLowerCase(),
+						id: this.contact.addressbook.id,
+						uid: this.contact.uid,
+					}))
+					if (response.status !== 200) {
+						throw new URIError('Download of social profile avatar failed')
+					}
+
+					// Fetch newly updated contact
+					await this.$store.dispatch('fetchFullContact', { contact: this.contact, forceReFetch: true })
+
+					// Update local clone
+					const contact = this.$store.getters.getContact(this.contact.key)
+					await this.$emit('updateLocalContact', contact)
+
+					// Notify user
+					OC.Notification.showTemporary(t('contacts', 'Avatar downloaded from social network'))
```

Comment:
I guess the avatar being loaded is enough, no need for another feedback?
```suggestion
```

Reply:
I noticed that in some cases the picture is refreshed but unchanged (to the eye). So that is why I introduced this info popup. I think it happens when the picture is converted in the contact and not byte-identical to the downloaded one. In these cases it looks as if nothing happened to the user, even though the image has been refreshed.

Reply:
AH, then we can get this in and if @jancborchardt disagree we'll remove later, this is no blocker ! :wink: